solidstart-tanstackquery-tailwind-modules/src/components/UserProfile/index.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  users,
+  star,
+  buildingOffice,
+  mapPin,
+  link,
+} from 'solid-heroicons/outline';
+import { Icon } from 'solid-heroicons';
+import { TwitterIcon } from '../Icons';
+import OrgList from './OrgList';
+import styles from './UserProfile.module.css';
+
+export interface UserProfileProps {
+  avatarUrl: string;
+  name: string;
+  username: string;
+  bio: string;
+  followers: number;
+  following: number;
+  starredRepos: number;
+  company: string;
+  location: string;
+  websiteUrl: string;
+  twitterUsername: string;
+  organizations: {
+    nodes: {
+      avatarUrl: string;
+      name: string;
+      login: string;
+    }[];
+  };
+}
+
+const UserProfile = (userProfileProps: UserProfileProps) => {
+  return (
```

Comment:
For the prop, can you check out solid js props
You can read here. 
https://www.solidjs.com/tutorial/props_defaults

Nothing wrong with your approach, but we wanna be more solid.
we have mergeProps and splitProps

Reply:
Likewise here, the props are direct. if you can outline the benefit for this particular use case.